subprojects/architecture-test/src/test/java/org/gradle/architecture/test/KotlinCompatibilityTest.java

```diff
@@ -0,0 +1,216 @@
+            if (superclassAccessors == null) {
+                return null;
+            }
+            if (!setters.isEmpty()) {
+                Set<JavaMethod> superclassGetters = superclassAccessors.stream().filter(Accessor::isGetter).map(Accessor::getMethod).collect(toSet());
+                if (!superclassGetters.isEmpty()) {
+                    return new Accessors(owningClass, propertyName, superclassGetters, setters);
+                }
+            } else {
+                Set<JavaMethod> superclassSetters = superclassAccessors.stream().filter(Accessor::isSetter).map(Accessor::getMethod).collect(toSet());
+                if (!superclassSetters.isEmpty()) {
+                    return new Accessors(owningClass, propertyName, getters, superclassSetters);
+                }
+            }
+            return null;
+        }
+
+        private Accessors(JavaClass owningClass, String propertyName, Set<JavaMethod> getters, Set<JavaMethod> setters) {
+            this.getters = getters;
+            this.setters = setters;
+            this.owningClass = owningClass;
+            this.propertyName = propertyName;
+        }
+
+        public boolean nullableIsSymmetric() {
+            Set<Boolean> gettersAreNullable = getters.stream().map(this::getterAnnotatedWithNullable).collect(toSet());
+            Set<Boolean> settersAreNullable = setters.stream().map(this::setterAnnotatedWithNullable).collect(toSet());
+            if (gettersAreNullable.size() > 1 || settersAreNullable.size() > 1) {
+                return false;
+            }
+            return gettersAreNullable.equals(settersAreNullable);
+        }
+
+        private boolean getterAnnotatedWithNullable(JavaMethod getter) {
+            return getter.isAnnotatedWith(Nullable.class);
+        }
+
+        private boolean setterAnnotatedWithNullable(JavaMethod setter) {
+            return Arrays.stream(setter.reflect().getParameterAnnotations()[0]).anyMatch(a -> a instanceof Nullable);
+        }
+
+        @Override
+        public String toString() {
+            return owningClass.getName() + "." + propertyName;
+        }
+    }
+
+    private static boolean isGetter(JavaMethod m, PropertyAccessorType accessorType) {
+        // We avoid using reflect, since that leads to class loading exceptions
+        return !m.getModifiers().contains(JavaModifier.STATIC)
+            && (accessorType == PropertyAccessorType.GET_GETTER || accessorType == PropertyAccessorType.IS_GETTER)
+            && m.getParameters().size() == 0
+            && (accessorType != PropertyAccessorType.IS_GETTER || m.getReturnType().isEquivalentTo(Boolean.TYPE) || m.getReturnType().isEquivalentTo(Boolean.class));
+    }
+
+    private static boolean isSetter(JavaMethod m, PropertyAccessorType accessorType) {
+        return !m.getModifiers().contains(JavaModifier.STATIC)
+            && accessorType == PropertyAccessorType.SETTER
+            && m.getParameters().size() == 1;
+    }
```

Comment:
:o: this will include methods with a non-void return type, I suppose it's ok

Reply:
That is our current logic to find setters for properties. I can also check for the void return type. WDYT?

Reply:
I'd say it's fine for this check to be a bit greedy and also consider setters that return something ; even if they are not javabean compliant at least the consistency of their nullability will be enforced. We can still make it more strict later if it proves to be problematic.